mage_ai/data_preparation/models/block/__init__.py

```diff
@@ -210,6 +211,7 @@ def __init__(
         name: str,
         uuid: str,
         block_type: BlockType,
+        block_color: BlockColor = None,
```

Comment:
I think it'd make more sense to just create a new Block for the custom block, and override some of the methods, kind of like the `SensorBlock`. https://github.com/mage-ai/mage-ai/blob/master/mage_ai/data_preparation/models/block/__init__.py#L1791

Reply:
ah ok